woocommerce/woocommerce-ios
# Update logic to download localized metadata from App Store Connect

### Description

This PR finally updates the localization automation for the ASC metadata to the latest version. It's basically a copy of [what WordPress iOS does](https://github.com/wordpress-mobile/WordPress-iOS/blob/e0464dd0bc3447f274d128a084e7b96d7c99add2/fastlane/lanes/localization.rb#L302-L345).

Originally, I planned on submitting a single PR for both `.strings` and metadata (see https://github.com/woocommerce/woocommerce-ios/pull/6737) but that work dragged along and I've become uncomfortable with the size of the diff itself and of the manual testing work. So, I've decided to split it.

I'm making this PR against the release branch because that's where we're most likely to use it next. @oguzkocer @jkmassel if you are not comfortable with this and would prefer to keep the release automation as is for this release, I can easily port the changes over to a PR against `trunk`.

I'll followup with a PR that removes the local code this makes redundant after this is merge, to keep the diff light on the release branch.

### Testing instructions

I created two test branches on top of this and run the localization automation using the old logic (`ios_update_metadata`) in one (#6880) and the new one in the other (#6881).

By inspecting the [commit generated in the new one](https://github.com/woocommerce/woocommerce-ios/pull/6881/commits/2baec6c7f971fcca7315f7153854ad427325a2cc) and by [comparing the two](https://github.com/woocommerce/woocommerce-ios/compare/localization-improvements-metadata-test-old...localization-improvements-metadata-test-new) we can verify that the new one behaves as expected:

- The `.txt` for the localized metadata are downloaded
- The `.txt` for the missing metadata are deleted, now including `name.txt`, `subtitle.txt` etc so that we can fallback to `fastlane/medata/default/` as the source of truth.

~~Unfortunately, at the time of testing there were no new `.strings` translations on GlotPress, so I couldn't verify my little hack to replace `ios_update_metadata` (see inline comments) worked.~~ Update: Luckily, there was an untranslated new Italian string. I translated it and you can see the automated commit it in both branches. 

---
